libsolidity/parsing/Parser.cpp

```diff
@@ -587,6 +594,7 @@ ASTPointer<ASTNode> Parser::parseFunctionDefinition(bool _freeFunction)
 				{Token::Receive, "receive function"},
 			}.at(m_scanner->currentToken());
 			name = make_shared<ASTString>(TokenTraits::toString(m_scanner->currentToken()));
+			nameLocation = currentLocation();
```

Comment:
Could we use this name location in some error message where we point to a variable? If so, perhaps it could be split out of this PR.